Unexpected error when using `idaes.core.util.testing.initialization_tester` twice on the same model test frame

On WaterTAP watertap/watertap/unit_models/tests/test_gac.py, tests encountered a failure for the following scenario:

1. model frame built and scaled
2. model frame initialization tested with `idaes.core.util.testing.initialization_tester` and **passed**
3. model solved and `check_optimal_termination` **passed**
4. costing block constructed with `WaterTAPCosting()`
5. model frame initialization tested with `idaes.core.util.testing.initialization_tester` and **failed**
6. model solved and `check_optimal_termination` **passed**

The error in step 5 is the following:
```
self = <idaes.core.base.process_block._ScalarGAC object at 0x0000022B5FFE0450>, name = '__dummy_equality_idx_index'
val = <pyomo.core.base.set.OrderedScalarSet object at 0x0000022B5FE94900>

    def add_component(self, name, val):
        """
        Add a component 'name' to the block.

        This method assumes that the attribute is not in the model.
        """
        #
        # Error checks
        #
        if not val.valid_model_component():
            raise RuntimeError(
                "Cannot add '%s' as a component to a block" % str(type(val)))
        if name in self._Block_reserved_words:
            raise ValueError("Attempting to declare a block component using "
                             "the name of a reserved attribute:\n\t%s"
                             % (name,))
        if name in self.__dict__:
>           raise RuntimeError(
                "Cannot add component '%s' (type %s) to block '%s': a "
                "component by that name (type %s) is already defined."
                % (name, type(val), self.name, type(getattr(self, name))))
E           RuntimeError: Cannot add component '__dummy_equality_idx_index' (type <class 'pyomo.core.base.set.OrderedScalarSet'>) to block 'fs.unit': a component by that name (type <class 'pyomo.core.base.set.OrderedScalarSet'>) is already defined.

..\..\..\anaconda3\envs\watertap-dev\lib\site-packages\pyomo\core\base\block.py:1056: RuntimeError
```

See also [active PR](https://github.com/watertap-org/watertap/blob/4c919e526a10c13516b6e933af7d84186ef3fbc5/watertap/unit_models/tests/test_gac.py#L430) for where `initialization_tester` would be at effectively line 430 to cause this failure.

--- a/idaes/core/util/testing.py
+++ b/idaes/core/util/testing.py
@@ -83,16 +83,17 @@ def initialization_tester(m, dof=0, unit=None, **init_kwargs):
     unit.__dummy_var = Var()
     unit.__dummy_equality = Constraint(expr=unit.__dummy_var == 5)
     unit.__dummy_inequality = Constraint(expr=unit.__dummy_var <= 10)
+    unit.__dummy_set = Set(initialize=[1])
 
     def deq_idx(b, i):
         return unit.__dummy_var == 5
 
-    unit.__dummy_equality_idx = Constraint([1], rule=deq_idx)
+    unit.__dummy_equality_idx = Constraint(unit.__dummy_set, rule=deq_idx)
 
     def dieq_idx(b, i):
         return unit.__dummy_var <= 10
 
-    unit.__dummy_inequality_idx = Constraint([1], rule=dieq_idx)
+    unit.__dummy_inequality_idx = Constraint(unit.__dummy_set, rule=dieq_idx)
 
     unit.__dummy_equality.deactivate()
     unit.__dummy_inequality.deactivate()
@@ -128,6 +129,7 @@ def initialization_tester(m, dof=0, unit=None, **init_kwargs):
     unit.del_component(unit.__dummy_inequality_idx)
     unit.del_component(unit.__dummy_equality_idx)
     unit.del_component(unit.__dummy_var)
+    unit.del_component(unit.__dummy_set)
 
 
 # -----------------------------------------------------------------------------

